src/octoprint/plugins/backup/__init__.py

```diff
@@ -129,6 +123,17 @@ def get_template_configs(self):
             {"type": "wizard", "name": gettext("Restore Backup?")},
         ]
 
+    ##~~ Utility Methods
+
+    def build_backup_filename(self):
+        if self._settings.global_get(["appearance", "name"]) == "":
+            backup_prefix = "octoprint"
+        else:
+            backup_prefix = self._settings.global_get(["appearance", "name"])
+        return "{}-backup-{}.zip".format(
+            backup_prefix, time.strftime(BACKUP_DATE_TIME_FMT)
```

Comment:
When backed up via CLI, the settings aren't injected so this fails.
Suggest adding a way that this can be specified in the arguments, like:
```python
def build_backup_filename(self, backup_prefix=None):
        if not backup_prefix:
                if self._settings.global_get(["appearance", "name"]) == "":
                        backup_prefix = "octoprint"  # Could put the default back above, so it can be used in CLI hook
                else:
                        backup_prefix = self._settings.global_get(["appearance", "name"])
        return "{}-backup-{}.zip".format(
            backup_prefix, time.strftime(BACKUP_DATE_TIME_FMT)
```
and then specify the default prefix in the CLI hook below. There might be another way to tell, but either way this needs fixing.